pkg/agent/conversation.go

```diff
@@ -205,11 +205,26 @@ func (a *Conversation) RunOneRound(ctx context.Context, query string) error {
 
 					s := toolCall.PrettyPrint()
 					a.UI.RenderOutput(ctx, fmt.Sprintf("  Running: %s\n", s), ui.Foreground(ui.ColorGreen))
-					if a.AsksForConfirmation && call.Arguments["modifies_resource"] == "no" {
-						confirm := a.UI.AskForConfirmation(ctx, "  Are you sure you want to run this command (Y/n)? ")
-						if !confirm {
-							a.UI.RenderOutput(ctx, "Sure.\n", ui.RenderMarkdown())
+					// Ask for confirmation only if SkipPermissions is false AND the tool modifies resources.
+					if !a.SkipPermissions && call.Arguments["modifies_resource"] != "no" {
+						confirmationPrompt := `  Do you want to proceed?
+  1) Yes
```

Comment:
Nit: y / Y / n might be more intuitive, but ... just a nit

Reply:
I hear you! I think UX wise, using a terminal UI library that supports mouse etc. will completely eliminate the need for typing and is far superior. For now, keeping this as-is, with the hope that we use TUI.